packages/lit-html/src/private-ssr-support.ts

```diff
@@ -33,12 +33,14 @@ export const _Σ = {
   markerMatch: p._markerMatch,
   HTML_RESULT: p._HTML_RESULT,
   getTemplateHtml: p._getTemplateHtml,
-  patchDirectiveResolve: (
-    directive: Directive,
-    fn: (this: Directive, values: unknown[]) => unknown
-  ) => {
-    directive._resolve = fn;
-  },
+  overrideDirectiveResolve: (
+    directiveClass: new (part: PartInfo) => Directive & {render(): unknown}
```

Comment:
we have the interface `DirectiveClass` in `directive.js`. Why do you need `render()` here?

Reply:
That was the only way I could make that mixin not complain about `directiveClass` not implementing the abstract render method.

The `Concrete<T>` thing doesn't work because it loses the "methodness" of `_$resolve`, and complains that you're assigning a function to a property in the subclass.